pkg/controller/core/capacity_controller.go

```diff
@@ -47,47 +55,121 @@ func NewCapacityReconciler(cache *capacity.Cache) *CapacityReconciler {
 //+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=capacities/finalizers,verbs=update
 
 func (r *CapacityReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	// No-op. All work is done in the event handler.
+	var capObj kueue.Capacity
+	if err := r.client.Get(ctx, req.NamespacedName, &capObj); err != nil {
+		// we'll ignore not-found errors, since there is nothing to do.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log := ctrl.LoggerFrom(ctx).WithValues("capacity", klog.KObj(&capObj))
+	ctx = ctrl.LoggerInto(ctx, log)
+	log.V(2).Info("Reconciling Capacity")
+
+	usage, workloads, err := r.cache.Usage(&capObj)
+	if err != nil {
+		log.Error(err, "Failed getting usage from capacity cache")
+		// This is likely because the capacity was recently removed and we didn't
+		// process that event yet.
+		return ctrl.Result{}, err
+	}
+	// Shallow copy enough for now.
+	oldStatus := capObj.Status
+	capObj.Status.UsedResources = usage
+	capObj.Status.RunningWorkloads = int32(workloads)
+	if !equality.Semantic.DeepEqual(oldStatus, capObj.Status) {
+		err = r.client.Status().Update(ctx, &capObj)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
 	return ctrl.Result{}, nil
 }
 
 func (r *CapacityReconciler) Create(e event.CreateEvent) bool {
-	c := e.Object.(*kueue.Capacity)
+	c, match := e.Object.(*kueue.Capacity)
+	if !match {
+		// No need to interact with the cache for other objects.
+		return true
+	}
 	log := r.log.WithValues("capacity", klog.KObj(c))
 	log.V(2).Info("Capacity create event")
 	ctx := ctrl.LoggerInto(context.Background(), log)
 	if err := r.cache.AddCapacity(ctx, c); err != nil {
 		log.Error(err, "Failed to add capacity to cache")
 	}
-	return false
+	return true
```

Comment:
please add a comment somewhere explaining that all event handlers will be followed by executing Reconcile.

Reply:
Done